### PR TITLE
Make upstart jobs more robust

### PR DESCRIPTION
**Pull request by 2opremio**

### CODE REVIEW

File: aws/ecs/packer/to-upload/run.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    fi
+}
+
+run_scope() {
+    while true; do
+	# verify that scope is not running
+	while is_container_running weavescope; do sleep 2; done
+	# launch scope
+	ARGS=""
+	if [ -e /etc/weave/scope.config ]; then
```
Comment (paulbellamy) [inline aws/ecs/packer/to-upload/run.sh]: Why `/etc/weave/scope.config` instead of `/etc/default/scope`?

File: aws/ecs/packer/to-upload/scope.conf [hunk 2]
```diff
@@ -1,23 +1,27 @@
+# Copyright 2014-2015 Weaveworks. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the
```
Comment (paulbellamy) [inline aws/ecs/packer/to-upload/scope.conf]: apache license?

File: aws/ecs/packer/template.json [hunk 3]
```diff
@@ -56,6 +60,8 @@
       "sudo mkdir /etc/weave",
       "sudo mv /home/ec2-user/peers.sh /etc/weave/peers.sh",
       "sudo chmod +x /etc/weave/peers.sh",
+      "sudo mv /home/ec2-user/peers.sh /etc/weave/run.sh",
```
Comment (2opremio) [inline aws/ecs/packer/template.json]: That's wrong, it should be `mv /home/ec2-user/run.sh /etc/weave/run.sh`

File: aws/ecs/packer/to-upload/scope.conf (hunk 2 above)
Comment (2opremio) [inline aws/ecs/packer/to-upload/scope.conf]: That's what we use everywhere

Comment (paulbellamy) [inline aws/ecs/packer/to-upload/scope.conf]: ok

File: aws/ecs/packer/to-upload/run.sh (hunk 1 above)
Comment (2opremio) [inline aws/ecs/packer/to-upload/run.sh]: Uhm, yep, you are probably right. Where we put the stuff calls for a different pull request though (since it also affects the ECS guide)

File: aws/ecs/packer/to-upload/run.sh [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+LOGGER_TAG=""
+
+is_container_running() {
+    CONTAINER_NAME=$1
+    [ "$(docker inspect -f '{{.State.Running}}' $1 2> /dev/null)" = true ]
+}
+
+succeed_or_die() {
+    if ! OUT=$($@ 2>&1); then
+	echo "error: $1 failed: $OUT" | logger -p local0.error -t $LOGGER_TAG
```
Comment (paulbellamy) [inline aws/ecs/packer/to-upload/run.sh]: logger will catch both stdout and stderr, yeah?

Comment (2opremio) [inline aws/ecs/packer/to-upload/run.sh]: Only stdout, but we capture both stdout and stderr in OUT

File: aws/ecs/packer/to-upload/run.sh [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+	# launch scope
+	ARGS=""
+	if [ -e /etc/weave/scope.config ]; then
+	    . /etc/weave/scope.config
+	fi
+	if [ -n "${SERVICE_TOKEN+x}" ]; then
+	    ARGS="$ARGS --service-token=$SERVICE_TOKEN"
+	fi
+	PEERS=$(succeed_or_die /etc/weave/peers.sh)
+	args="$ARGS 127.0.0.1 $PEERS"
```
Comment (paulbellamy) [inline aws/ecs/packer/to-upload/run.sh]: Doesn't peers include my own IP?

File: aws/ecs/packer/to-upload/run.sh [hunk 6]
```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -eu
+
+LOGGER_TAG=""
+
+is_container_running() {
+    CONTAINER_NAME=$1
+    [ "$(docker inspect -f '{{.State.Running}}' $1 2> /dev/null)" = true ]
```
Comment (2opremio) [inline aws/ecs/packer/to-upload/run.sh]: Either remove CONTAINER_NAME or use it instead of $1

File: aws/ecs/packer/to-upload/run.sh (hunk 4 above)
Comment (paulbellamy) [inline aws/ecs/packer/to-upload/run.sh]: ah, yeah.

File: aws/ecs/packer/to-upload/run.sh (hunk 1 above)
Comment (paulbellamy) [inline aws/ecs/packer/to-upload/run.sh]: Ok. created: https://github.com/weaveworks/integrations/issues/9

Comment (2opremio) [inline aws/ecs/packer/to-upload/run.sh]: Thanks!

File: aws/ecs/packer/to-upload/run.sh (hunk 4 above)
Comment (paulbellamy) [inline aws/ecs/packer/to-upload/run.sh]: We say `$1 failed`, but we called `$@`

File: aws/ecs/packer/to-upload/run.sh (hunk 5 above)
Comment (2opremio) [inline aws/ecs/packer/to-upload/run.sh]: No, we exclude it explicitly since we don't want to use it in the case of Weave.

**Comment by 2opremio** [comment]
Fixes #8
